docs/proposals/PROPOSAL_E.md

```diff
@@ -166,23 +166,22 @@ When upgrading existing registries, the following manifests MUST be scanned for
 
 Clients SHALL NOT expect manifests uploaded before the [referrers API](#referrers-api) is available on that registry, without using a [digest tag](#digest-tags), will be included in future API responses.
 
-#### Digest Tags
+#### Digest Tag
 
-For registries that do not support the `referrers` API, a tag MUST be pushed for any manifest containing a `refers` descriptor with the following syntax:
+For registries that do not support the `referrers` API, a tag MUST be pushed containing an index of all manifests that refer to the specified manifest, matching the API response above.
```

Comment:
There's an alternative approach where referring things are still attached by digest tags, but are pre-filtered by type using conventional suffixes, e.g., `:sha256-abcdef.sig`. This has some downsides, like having to document the convention of what `.sbom` means, but this seems like a tractable problem at least. It also means people are free to add `.catpictures` for their own use cases if they want to.

Benefits:
- the fallback still allows some rudimentary form of filtering by type
- the convention is arbitrarily extensible
- users don't have to list tags to get attachments of known types (`.sbom`, `.sig`)
- this convention is already widely used in the wild today, by `cosign` etc.

Drawbacks:
- we have to document the conventional types/suffixes
- clients that want to list _all_ things of any arbitrary types (`.catpictures`) will have to list tags

Reply:
One thing I was looking at was how to avoid stepping on the existing cosign signatures since they already exist in repositories now with rolling that expects to find a specific format within the manifest. If we did step on the same tag syntax, we'll need clients to know if and how to parse the response from both this proposal and existing cosign tooling.

Reply:
+1 here, happy to change cosign in any way required to make this work.